crates/sui-core/src/consensus_handler.rs

```diff
@@ -53,11 +62,75 @@ impl<T> ConsensusHandler<T> {
             checkpoint_service,
             transaction_manager,
             parent_sync_store,
+            low_scoring_authorities: scores_per_authority,
+            committee,
             metrics,
         }
     }
 }
 
+/// Updates list of authorities that are deemed to have low reputation scores by consensus
+/// these may be lagging behind the network, byzantine, or not reliably participating for any reason.
+/// We want to ensure that the remaining set of validators once we exclude the low scoring authorities
+/// is including enough stake for a quorum, at the very least. It is also possible that no authorities
+/// are particularly low scoring, in which case this will result in storing an empty list.
+fn update_low_scoring_authorities(
+    low_scoring_authorities: Arc<DashMap<AuthorityName, u64>>,
+    committee: Committee,
+    reputation_scores: ReputationScores,
+) {
+    if !reputation_scores.final_of_schedule {
+        return;
+    }
+    let mut score_list = vec![];
+    for val in reputation_scores.scores_per_authority.values() {
+        score_list.push(*val as f64);
+    }
+
+    let median = statistical::median(&score_list);
+    let mut deviations = vec![];
+    let mut abs_deviations = vec![];
+    for (i, _) in score_list.clone().iter().enumerate() {
+        deviations.push((score_list[i] - median) * -1.0);
```

Comment:
can we exclude everyone above median and then just work with one vector?

Reply:
Why are you inverting the deviations ? (multiplying by -1)